### PR TITLE
test: add tests for prepareEmailSubmission

### PR DESCRIPTION
**Pull request by mantariksh**
This PR adds tests for #752.

### CODE REVIEW

File: src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+  generateNewAttachmentResponse,
+  generateNewCheckboxResponse,
+  generateNewSingleAnswerResponse,
+  generateNewTableResponse,
+} from 'tests/unit/backend/helpers/generate-form-data'
+
+import { ProcessedSingleAnswerResponse } from '../../submission.types'
+import * as EmailSubmissionService from '../email-submission.service'
+
+const ALL_SINGLE_SUBMITTED_RESPONSES = [
```
Comment (karrui) [inline src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts:15]: should filter through all fields to get the single fields so when we add new fields these tests will automatically cover them

File: src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  generateNewSingleAnswerResponse(BasicField.LongText),
+  generateNewSingleAnswerResponse(BasicField.Dropdown),
+  generateNewSingleAnswerResponse(BasicField.YesNo),
+  generateNewSingleAnswerResponse(BasicField.Radio),
+  generateNewSingleAnswerResponse(BasicField.Date),
+  generateNewSingleAnswerResponse(BasicField.Rating),
+  generateNewSingleAnswerResponse(BasicField.Nric),
+]
+
+const generateSingleAnswerJson = (response: ProcessedSingleAnswerResponse) =>
```
Comment (karrui) [inline src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts:31]: should these utils be in helpers too?

File: src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+        new Set(),
+      )
+
+      const question = response.question
+      const firstRow = response.answerArray[0].join(',')
+      const secondRow = response.answerArray[1].join(',')
+
+      expect(emailData).toEqual({
+        jsonData: [
+          { question: `[table] ${question}`, answer: firstRow },
```
Comment (karrui) [inline src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts:115]: should we use the constants defined?

File: src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+        autoReplyData: [
+          { question, answerTemplate: [firstRow] },
+          { question, answerTemplate: [secondRow] },
+        ],
+        formData: [
+          {
+            question: `[table] ${question}`,
+            answer: firstRow,
+            answerTemplate: [firstRow],
+            fieldType: 'table',
```
Comment (karrui) [inline src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts:127]: use the enum?

File: src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+    })
+
+    it('should prefix MyInfo-verified fields with [MyInfo] only in form data', () => {
+      const response = generateNewSingleAnswerResponse(BasicField.ShortText, {
+        myInfo: { attr: MyInfoAttribute.Name },
+      })
+
+      const emailData = EmailSubmissionService.createEmailData(
+        [response],
+        new Set([response._id]),
```
Comment (karrui) [inline src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts:297]: can we improve this test case to test for multiple responses? or even add another test case for duplicate myinfo field types to verify the recent bug does not occur

File: tests/unit/backend/helpers/generate-form-data.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { ObjectID } from 'bson'
+import { ObjectID, ObjectId } from 'bson'
```
Comment (karrui) [inline tests/unit/backend/helpers/generate-form-data.ts:1]: nit: can remove the `ID` one in favor of `Id`

**Review by karrui — APPROVED** [review]
lgtm with suggestions

File: tests/unit/backend/helpers/generate-form-data.ts (hunk 6 above)
Comment (mantariksh) [reply tests/unit/backend/helpers/generate-form-data.ts:1]: oh oops RIP lol
